File: CSharp/SpecStudioExamplesTestsCSharp/production/OrderItemCollection.cs
namespace production
{
    using System.Collections.Generic;

    public class OrderItemCollection
    {
        public const int Minimum = 0;
        public const int Maximum = 100;

        private readonly List<OrderItem> _items = new();

        public void Add(OrderItem item) => _items.Add(item);

        public bool Delete(OrderItem item) => _items.Remove(item);

        public IReadOnlyList<OrderItem> Read() => _items.AsReadOnly();

        public bool Update(OrderItem oldItem, OrderItem newItem)
        {
            int idx = _items.IndexOf(oldItem);
            if (idx < 0) return false;
            _items[idx] = newItem;
            return true;
        }

        public int Count => _items.Count;
    }
}

File: Java/SpectableJavaExamplesTest/src/main/java/production/OrderItemCollection.java
package production;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class OrderItemCollection {
    public static final int MINIMUM = 0;
    public static final int MAXIMUM = 100;

    private final List<OrderItem> items = new ArrayList<>();

    public void add(OrderItem item) {
        items.add(item);
    }

    public boolean delete(OrderItem item) {
        return items.remove(item);
    }

    public List<OrderItem> read() {
        return Collections.unmodifiableList(items);
    }

    public boolean update(OrderItem oldItem, OrderItem newItem) {
        int index = items.indexOf(oldItem);
        if (index < 0) return false;
        items.set(index, newItem);
        return true;
    }

    public int size() {
        return items.size();
    }

    public Dollar computeTotal() {
        Dollar total = new Dollar("0");
        for (OrderItem item : items) {
            total = total.plus(item.itemTotal);
        }
        return total;
    }
}
